lib/mono/cli/publish.rb

```diff
@@ -184,6 +193,16 @@ def package_promoter
         @package_promoter ||=
           PackagePromoter.new(dependency_tree, :prerelease => prerelease)
       end
+
+      def existing_tags(changed_packages)
+        changed_packages.map(&:next_tag).reject do |tag|
+          run_command(
+            "git tag -l #{tag}",
+            :capture => true,
+            :print_command => false
+          ).strip.empty?
+        end
```

Comment:
```suggestion
        tags = run_command(
          "git tag --list",
          :capture => true,
          :print_command => false
        ).split("\n")
        changed_packages.map(&:next_tag).select do |tag|
          tags.include?(tag)
        end
```
A bit more efficient. We don't need to run it separately per package.

Reply:
Is it more efficient? I'd expect the output of `git tag --list` to usually be quite long (lots of previous releases) while the amount of packages to run `git tag -l #{tag}` for would be relatively small (usually once)

But it's likely that the time spent shelling out several times greatly outweighs the time spent listing all releases. I'll have a look at your solution.

Reply:
Hm, best of both worlds?

```ruby
next_tags = changed_packages.map(&:next_tag).join(" ")
run_command(
  "git tag --list #{next_tags}",
  :capture => true,
  :print_command => false
).strip().split("\n")
```

Reply:
Looks good 👍

Reply:
Done ✅ Thank you!